### PR TITLE
Disable CI for draft PRs

### PR DESCRIPTION
**Pull request by anthayes92**
**Context:**
Tooling Team has requested that all CI checks for draft PRs across all PennyLane repos are to be disabled to free up GH runner resources. [SC Story](https://app.shortcut.com/xanaduai/story/66346/disable-ci-for-pl-draft-prs).

**Description of the Change:**
Conditions for checking whether a PR is in draft state have been added to existing workflows.

**Benefits:**
Reduces GH runner usage.

**Possible Drawbacks:**
Some development steps are deferred to PRs in "ready for review" state only.

**Related GitHub Issues:**
N/A

### Verification:
Created this PR as draft, only lightweight CI checks use GitHub runners (formatting, changelog reminder). When marked as "ready for review" the CI checks ran.

Screenshot from 2024-08-15 15-18-11

### CODE REVIEW

**Review by albi3ro — APPROVED** [review]
🎉

File: .github/workflows/tests-gpu.yml [hunk 1]
```diff
@@ -28,6 +29,7 @@ jobs:
       ${{
         github.event_name == 'push' ||
         (
+          github.event.pull_request.draft == false &&
           github.event_name == 'pull_request' &&
```
Comment (Mandrenkov) [inline .github/workflows/tests-gpu.yml:33]: # Note
I'm a bit surprised the ordering here doesn't matter but 🤷. 
```suggestion
          github.event_name == 'pull_request' &&
          github.event.pull_request.draft == false &&
```

**Review by Mandrenkov — APPROVED** [review]

**Comment by mlxd** [comment]
Hey @anthayes92 any reason this is blocked? Can this be merged?

**Comment by anthayes92** [comment]
>Hey @anthayes92 any reason this is blocked? Can this be merged?

An additional requirement came in about adding a label to run tests in draft mode. Out of scope of the original story but makes sense to do in the same PR, just need to find the time to do this.

Alternatively a temporary solution is to switch to "ready for review " to run tests and back to "draft " when desired, but part of the problem seems to be communicating this to all the devs.

**Comment by Alex-Preciado** [comment]
I have removed the "do not merge ⚠️" label. We can add a label to run in draft move in a follow up PR if necessary Let's merge this today!

**Pull request merged**